frontend/src/features/dsPreview/state/dsPreviewState.ts

```diff
@@ -0,0 +1,31 @@
+import { createReducer } from '@reduxjs/toolkit'
+
+import { RootState } from '../../../store/store';
+
+export const selectDsPreview = (state: RootState): any => state.dsPreview.preview
+
+export const selectIsDsPreviewLoading = (state: RootState): boolean => state.dsPreview.loading
+
+export interface DsPreviewState {
+  preview: any
```

Comment:
we need a real type here. `Dataset` should work. Also, should be fine to make this property optional